src/chunk.h

```diff
@@ -46,6 +46,13 @@ typedef struct Chunk
 	ChunkConstraints *constraints;
 } Chunk;
 
+typedef struct ChunkStub
```

Comment:
I think this needs a comment explaining what it is conceptually. To my eye it seems like it identifies a subspace of a hyperspace which _could_ be covered by a single chunk.